src/app/Features/CheckInconsistencies/Tautology/index.tsx

```diff
@@ -162,124 +177,141 @@ export class TautologyInconsistencies extends BaseFeature {
   };
 
   private getProblemOutputConditionsHtmlList = (
-    outputCondictions: any[]
-  ): any => {
+    outputCondictionsDetails: ConditionActionProblemDetail[]
+  ): React.ReactElement => {
     return (
       <ul
         style={{ fontSize: '0.875rem', marginTop: '0.5rem', color: '#607b99' }}
       >
-        {outputCondictions.map((oc, index) => (
+        {outputCondictionsDetails.map((oc, index) => (
           <li key={index}>
             {oc.blockName} - {oc.variable}
           </li>
         ))}
       </ul>
     );
   };
-}
 
-const getActionsWithExecuteConditions = (block: any): any => {
-  return getAllActions(block).filter(hasExecuteCondition);
-};
-
-const getAllActions = (block: any): any => {
-  const blockName = block.$title;
-
-  return [
-    ...block.$enteringCustomActions.map((action) =>
-      insertBlockNameOnAction(action, blockName)
-    ),
-    ...block.$leavingCustomActions.map((action) =>
-      insertBlockNameOnAction(action, blockName)
-    ),
-  ];
-};
-
-const insertBlockNameOnAction = (action: any, blockName: string): any => {
-  return { ...action, blockName: blockName };
-};
-
-const hasExecuteCondition = (action: any): boolean =>
-  action.conditions.length > 0;
-
-const hasTautologyOnCondition = (condition: any): boolean =>
-  condition.comparison === NOT_EQUAL_CONDITION && condition.values.length > 1;
-
-const getActionsCondictionsWithTautologyDetail = (
-  action: any
-): ConditionActionProblemDetail[] => {
-  const condictionsWithTautology = [];
-  for (const condition of action.conditions) {
-    if (hasTautologyOnCondition(condition)) {
-      condictionsWithTautology.push({
-        blockName: action.blockName,
-        actionName: action.$title,
-        variable: condition.variable,
-      });
-    }
+  private getActionsWithExecuteConditions = (block: BlipFlowBlock): BlipAction[] => {
+    return this.getAllActions(block).filter(this.hasExecuteCondition);
+  };
+
+  private getAllActions = (block: BlipFlowBlock): BlipAction[] => {
```

Comment:
in the commit related to the auto tag feature this function was created in the utils file, i think its important to map this cases to future refactors